packages/cli/src/commands/integration/remove.ts

```diff
@@ -0,0 +1,358 @@
+  );
+
+  if (!targetedResource) {
+    client.output.error(`No resource ${chalk.bold(resourceName)} found.`);
+    return 0;
+  }
+
+  if (args.length === 3) {
+    if (flags['--unlink-all']) {
+      client.output.error(
+        'Cannot specify a project when using the `--unlink-all` flag.'
+      );
+      return 1;
+    }
+
+    const unlinkProjectResults = await handleUnlinkProject(
+      client,
+      targetedResource,
+      args[2],
+      !!flags['--yes']
+    );
+    if (unlinkProjectResults !== undefined) {
+      return unlinkProjectResults;
+    }
+  } else if (flags['--unlink-all']) {
+    const unlinkAllResults = await handleUnlinkAllProjects(
+      client,
+      targetedResource,
+      !!flags['--yes']
+    );
+    if (unlinkAllResults !== undefined) {
+      return unlinkAllResults;
+    }
+  }
+
+  if (flags['--delete']) {
+    const deleteResult = await handleDeleteResource(
+      client,
+      team,
+      targetedResource,
+      !!flags['--yes']
+    );
+    if (deleteResult !== undefined) {
+      return deleteResult;
+    }
+  }
+
+  return 0;
+}
+
+// $ vc integration remove [resource-name] [project]
+async function handleUnlinkProject(
+  client: Client,
+  resource: Resource,
+  projectName: string,
+  skipConfirmation: boolean
+): Promise<number | undefined> {
+  const project = resource.projectsMetadata?.find(
+    project => projectName === project.name
+  );
+  if (!project) {
+    client.output.log(
+      `Could not find project ${chalk.bold(projectName)} linked to resource ${chalk.bold(resource.name)}.`
+    );
+    return 0;
+  }
+
+  if (
+    !skipConfirmation &&
+    !(await confirmUnlinkProject(client, resource, project))
+  ) {
+    client.output.log('Canceled');
+    return 0;
+  }
+
+  try {
+    client.output.spinner('Unlinking resource…', 500);
+    await disconnectResourceFromProject(client, resource, project);
+    client.output.success(
+      `Unlinked ${chalk.bold(project.name)} from ${chalk.bold(resource.name)}`
+    );
+    resource.projectsMetadata = resource.projectsMetadata?.filter(
+      project => projectName !== project.name
+    );
+  } catch (error) {
+    client.output.error(
+      `A problem occurred while unlinking: ${(error as Error).message}`
+    );
+    return 1;
+  }
+
+  return;
+}
+
+// $ vc integration remove [resource-name] --unlink-all
+async function handleUnlinkAllProjects(
+  client: Client,
+  resource: Resource,
+  skipConfirmation: boolean
+): Promise<number | undefined> {
+  if (resource.projectsMetadata?.length === 0) {
+    client.output.log(
+      `${chalk.bold(resource.name)} has no projects to unlink.`
+    );
+  } else {
+    if (
+      !skipConfirmation &&
+      !(await confirmUnlinkAllProjects(client, resource))
+    ) {
+      client.output.log('Canceled');
+      return 0;
+    }
+
+    const unlinkApiCalls =
+      resource.projectsMetadata?.map(connection => {
+        return disconnectResourceFromProject(client, resource, connection);
```

Comment:
I don't suppose we could make an API endpoint that does this in bulk?  I don't love loop interactions like this that can leave the system in a half committed state. If not, then let's forge ahead but I'd love to see a message telling the customer the current state and what to do next.

Reply:
API endpoint to perform the intended action in a single call: https://github.com/vercel/api/pull/33142

Reply:
New API endpoint works, is merged, and deployed to production. I've adjusted this route to hit that endpoint instead of sending a burst of requests.